src/cdk/v2/destinations/intercom/utils.js

```diff
@@ -503,6 +504,89 @@ const addOrUpdateTagsToCompany = async (message, destination, id) => {
   );
 };
 
+/**
+ * Api call to get company id provided by intercom
+ * Ref doc v1: https://developers.intercom.com/docs/references/1.4/rest-api/companies/view-a-company
+ * Ref doc v2: https://developers.intercom.com/docs/references/2.10/rest-api/api.intercom.io/companies/retrievecompany
+ * @param {*} company
+ * @param {*} destination
+ * @returns
+ */
+const getCompanyId = async (company, destination) => {
+  if (!company.id && !company.name) return undefined;
+  const { apiVersion } = destination.Config;
+  const headers = getHeaders(destination, apiVersion);
+  const baseEndPoint = getBaseEndpoint(destination);
+
+  const queryParam = company.id ? `company_id=${company.id}` : `name=${company.name}`;
+  const endpoint = `${baseEndPoint}/companies?${queryParam}`;
+
+  const statTags = {
+    destType: 'intercom',
+    feature: 'transformation',
+    endpointPath: '/companies',
+    requestMethod: 'POST',
+    module: 'router',
+  };
+
+  const response = await httpGET(
+    endpoint,
+    {
+      headers,
+    },
+    statTags,
+  );
+
+  const processedResponse = processAxiosResponse(response);
+  if (isHttpStatusSuccess(processedResponse.status)) {
+    return processedResponse.response.id;
+  }
+  intercomErrorHandler('Unable to get company id due to', processedResponse);
+  return undefined;
+};
+
+/**
+ * Api call to detach contact and company for intercom api version v2 (version 2.10)
+ * Ref doc: https://developers.intercom.com/docs/references/2.10/rest-api/api.intercom.io/contacts/detachcontactfromacompany
+ * @param {*} contactId
+ * @param {*} message
+ * @param {*} destination
+ * @returns
+ */
+const detachContactAndCompany = async (contactId, message, destination) => {
+  const company = message?.traits?.company || message?.context?.traits?.company;
+  if (!company || !company.remove) return;
+
+  // Detaching company when company.remove : true
+  const companyId = await getCompanyId(company, destination);
+  if (!companyId) return;
+
+  const headers = getHeaders(destination);
+  const baseEndPoint = getBaseEndpoint(destination);
+  const endpoint = `${baseEndPoint}/contacts/${contactId}/companies/${companyId}`;
+
+  const statTags = {
+    destType: 'intercom',
+    feature: 'transformation',
+    endpointPath: 'contacts/{contact_id}/companies/{id}',
```

Comment:
better to skip dynamic params in endpointPath, this will increase the cordinality